zooniverse/front-end-monorepo
# Add invert button to image toolbar

## Package
- lib-classifier

## Linked Issue and/or Talk Post
- closes #1716 

## Describe your changes
- adds image invert button to image toolbar
- adds `invert_subject` to workflow configuration model
- conditionally show invert button per workflow configuration
- add invert property to subject viewer model
- add invert styling to image viewer per subject viewer invert property
- refactor with react-testing-library

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - https://local.zooniverse.org:8080/?project=1952&workflow=3594&demo=true 
- What user actions should my reviewer step through to review this PR?
  - click Invert button, see image colors inverted
  - click different toolbar button, invert button remains actively styled (like annotation button)
  - click Invert button again, see image colors revert to original
- Which storybook stories should be reviewed?
  - Invert button - http://localhost:6006/?path=/story/image-toolbar-invertbutton--default&globals=locale:en;locales.en:English;locales.test:Test+Language
  - Image toolbar with Invert button - http://localhost:6006/?path=/story/image-toolbar-image-toolbar--default&globals=locale:en;locales.en:English;locales.test:Test+Language
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes - update the PFE FEM Lab to enable the related workflow setting checkbox

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)
